docs/core.md

```diff
@@ -0,0 +1,361 @@
+# Core library
+
+The *Core library* implements common assertions about the exit code and output
+captured with the most recent `run()` command.
+
+
+## `flunk()`
+
+Display an error message and fail. This function provides a convenient way to
+report failure in arbitrary situations. You can use it to implement your own
+helpers when the ones available do not meet your needs. Other functions use it
+internally as well.
+
+```bash
+@test 'flunk()' {
+  flunk 'this test always fails'
+}
+```
+
+The message can also be specified on the standard input.
+
+```bash
+@test 'flunk() with pipe' {
+  echo 'this test always fails' | flunk
+}
+```
+
+This function always fails and simply outputs the given message.
+
+```
+this test always fails
+```
+
+
+## `assert()`
+
+Fail if the given expression evaluates to false.
+
+```bash
+@test 'assert()' {
+  local value=1
+  assert [ "$value" -eq 0 ]
```

Comment:
We'll need to be clear that you can't use `[[` here because it's not a simple command.